### PR TITLE
Fix hd string path parsing

### LINKED ISSUE #1014: Bug in bitcore.HDPrivateKey.isValidPath
```
var isValid = bitcore.HDPrivateKey.isValidPath('m/12asd'); //  returns true;
```

And you can also derive it
```
var xkey = new bitcore.HDPrivateKey();
var derived1 = xkey.derive('m/12asd');
var derived2 = xkey.derive('m/2147483660');

assert.equals(derived1, derived2);
```

### PR DESCRIPTION
**Pull request by yemel**
Closes #1014

### CODE REVIEW

**Comment by eordano** [comment]
LGTM

File: lib/hdprivatekey.js [hunk 1]
```diff
@@ -106,8 +106,13 @@ HDPrivateKey._getDerivationIndexes = function(path) {
...
-    index += step != index.toString() ? HDPrivateKey.Hardened : 0;
+    var index = step ? +step : NaN;
+
+    var isHardened = isNaN(index) && step[step.length-1] == "'";
+    if (isHardened) {
+      index = (+(step.slice(0, -1))) + HDPrivateKey.Hardened;
+    }
+
     return index;
   });
```
Comment (maraoz) [inline lib/hdprivatekey.js]: This code is way too complicated for what it does. I don't want to be a PITA, but what can you consider rewriting it in a simpler way? 
I'm talking about the whole `_getDerivationIndexes` function

Comment (maraoz) [inline lib/hdprivatekey.js]: or maybe we can do it in another PR if you want

Comment (eordano) [inline lib/hdprivatekey.js]:
```javascript
  var privateValidPath = function(path) {
    return !!(/^[mM][']?(\/[0-9]+[']?)*[/]?$/.exec(path));
  };
  var publicValidPath = function(path) {
    return !!(/^[mM](\/[0-9]+)*[/]?$/.exec(path));
  };
```

**Pull request merged**